src/test/java/io/r2dbc/postgresql/ArrayAggIntegrationTests.java

```diff
@@ -0,0 +1,52 @@
+package io.r2dbc.postgresql;
+
+import io.r2dbc.postgresql.api.PostgresqlConnection;
+import io.r2dbc.postgresql.api.PostgresqlResult;
+import io.r2dbc.postgresql.codec.Json;
+import io.r2dbc.postgresql.util.PostgresqlServerExtension;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+import org.springframework.jdbc.core.JdbcOperations;
+import reactor.test.StepVerifier;
+
+public class ArrayAggIntegrationTests {
+  @RegisterExtension
+  static final PostgresqlServerExtension SERVER = new PostgresqlServerExtension();
+
+  private final PostgresqlConnectionConfiguration configuration = PostgresqlConnectionConfiguration.builder()
+      .database(SERVER.getDatabase())
+      .host(SERVER.getHost())
+      .port(SERVER.getPort())
+      .password(SERVER.getPassword())
+      .username(SERVER.getUsername())
+      .forceBinary(true)
+      .build();
+
+  private final PostgresqlConnectionFactory connectionFactory = new PostgresqlConnectionFactory(this.configuration);
+
+  @Test
+  void shouldReadArrayAgg() {
+
+    JdbcOperations jdbcOperations = SERVER.getJdbcOperations();
+    jdbcOperations.execute("DROP TABLE IF EXISTS my_table;");
+    jdbcOperations.execute("CREATE TABLE my_table (t text)");
+
+    PostgresqlConnection connection = this.connectionFactory.create().block();
+
+    connection.createStatement("INSERT INTO my_table (t) VALUES('hello'), ('world'), (null)")
+        .execute()
+        .flatMap(PostgresqlResult::getRowsUpdated)
+        .as(StepVerifier::create)
+        .expectNextCount(1)
+        .verifyComplete();
+
+    connection.createStatement("SELECT array_agg(t) FROM my_table")
+        .execute()
+        .flatMap(it -> it.map((row, rowMetadata) -> row.get("array_agg")))
+        .as(StepVerifier::create)
+        .expectNext(new Object[]{"hello", "world", null})
+        .verifyComplete();
+
+    connection.close().block();
```

Comment:
@mp911de I'm not sure how to do this bit?

Reply:
You can subclass `AbstractIntegrationTests`. That will give you access to a `connection` field so you don't need to worry about the connection lifecycle. Other than that, `close.block()` (or `close().as(StepVerifier::create).verifyComplete()`) are fine.

Reply:
FYI, I copied `JsonIntegrationTests` which apparently don't subclass `AbstractIntegrationTests`

Reply:
I filed #364 to fix `JsonIntegrationTests`. I think, previously we had no config hook to force binary transfer.